### PR TITLE
Add the pages used for sign in.

### PR DESCRIPTION
**Pull request by alexdesi**
### Context
We need to build the sign-in workflow, which makes use of the magic link.

### Changes proposed in this pull request

This adds the pages, that will be used in the sign-in workflow.

### Link to Trello card

https://trello.com/c/0wmsvsgX/64-implement-sign-in-using-magic-links

### CODE REVIEW

**Comment by tvararu** [comment]
@alexdesi could you attach screenshots? ✨

File: app/views/candidate_interface/sign_in/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    <h1 class="govuk-heading-xl">Sign in</h1>
+
+    <p class="govuk-body-l">Enter the email address you used to register with, and we will send you a link to sign in.</p>
+
+    <%= form_with model: @candidate, url: candidate_interface_sign_in_path, builder: GOVUKDesignSystemFormBuilder::FormBuilder, html: { novalidate: true } do |f| %>
+      <%= f.govuk_error_summary %>
+
+      <%= f.govuk_text_field :email_address, label: { text: t('authentication.sign_up.email_address.label')}, hint_text: t('authentication.sign_up.email_address.hint_label'), type: :email %>
+
+      <%= f.submit t('authentication.sign_up.button'), class: 'govuk-button', 'data-module': 'govuk-button', 'data-prevent-double-click': true %>
```
Comment (tvararu) [inline app/views/candidate_interface/sign_in/new.html.erb]: What is `data-prevent-double-click`? Part of the form elements library?

File: spec/system/candidate_interface/candidate_signing_in_spec.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+      fill_in t('authentication.sign_up.email_address.label'), with: 'new_candidate@example.com'
+      # click_on t('authentication.sign_up.button')
+    end
+
+    it 'sees the the sign in page' do
+      expect(page).to have_content 'Enter the email address you used to register with'
+    end
+
+    it 'see the "Check your email" page' do
```
Comment (tvararu) [inline spec/system/candidate_interface/candidate_signing_in_spec.rb]:
```suggestion
    it 'sees the "Check your email" page' do
```

**Comment by alexdesi** [comment]
I am rebasing this, please do not review yet :)

File: app/views/candidate_interface/sign_in/new.html.erb (hunk 1 above)
Comment (alexdesi) [reply app/views/candidate_interface/sign_in/new.html.erb]: it is part of the form builder:
https://github.com/DFE-Digital/govuk_design_system_formbuilder/blob/82bb7cbf440835338bab688b4af38d139816b9d4/lib/govuk_design_system_formbuilder/elements/submit.rb#L9

**Comment by alexdesi** [comment]
Screenshot 2019-09-26 at 12 12 22
Screenshot 2019-09-26 at 12 12 40

File: spec/system/candidate_interface/candidate_signing_in_spec.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+describe 'A candidate signing in' do
+  context 'who successfully signs in' do
+    before do
+      visit '/'
+      click_on 'sign in'
+
+      fill_in t('authentication.sign_up.email_address.label'), with: 'new_candidate@example.com'
+      # click_on t('authentication.sign_up.button')
```
Comment (duncanjbrown) [inline spec/system/candidate_interface/candidate_signing_in_spec.rb]: Could this be safely removed?

File: spec/system/candidate_interface/candidate_signing_in_spec.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+describe 'A candidate signing in' do
+  context 'who successfully signs in' do
```
Comment (duncanjbrown) [inline spec/system/candidate_interface/candidate_signing_in_spec.rb]: This context doesn't necessarily apply to the `sees the sign in page` spec. Could the `before` live at the top level, and then this context wrap the `sees the "Check your email" page`?

File: spec/system/candidate_interface/candidate_signing_in_spec.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      # click_on t('authentication.sign_up.button')
+    end
+
+    it 'sees the the sign in page' do
+      expect(page).to have_content 'Enter the email address you used to register with'
+    end
+
+    it 'sees the "Check your email" page' do
+      fill_in t('authentication.sign_in.email_address.label'), with: 'new_candidate@example.com'
+      click_on t('authentication.sign_up.button')
```
Comment (duncanjbrown) [inline spec/system/candidate_interface/candidate_signing_in_spec.rb]: This looks wrong — could this be `sign_in.button`? The i18n duplication doesn't matter

File: config/locales/application_form.yml [hunk 6]
```diff
@@ -1,3 +1,4 @@
 en:
   application_form:
     begin_button: Start now
+
```
Comment (duncanjbrown) [inline config/locales/application_form.yml]: Stray newline

File: spec/system/candidate_interface/candidate_signing_in_spec.rb (hunk 4 above)
Comment (alexdesi) [reply spec/system/candidate_interface/candidate_signing_in_spec.rb]: I agree! It should be good now:
```A candidate signing in
  sees the sign in page
  who successfully signs in
    sees the "Check your email" page
```

File: spec/system/candidate_interface/candidate_signing_in_spec.rb (hunk 5 above)
Comment (alexdesi) [reply spec/system/candidate_interface/candidate_signing_in_spec.rb]: residue of copy&past :)

**Comment by alexdesi** [comment]
I've done various improvements, should be good now.

File: app/views/candidate_interface/start_page/show.html.erb [hunk 7]
```diff
@@ -23,6 +24,10 @@
       <li>grade 4 (C) or above in English and maths GCSEs</li>
       <li>grade 4 (C) or above in GCSE science if you want to teach primary</li>
     </ul>
+    <p>Before you start a teacher training course you’ll also need to confirm you have the health and physical capacity to start training.</p>
+
+    <h2 class="govuk-heading-m">If you don’t have a degree</h2>
+    <p class="govuk-body">You can teach in further education (age 14 to adult), or study for a degree leading to qualified teacher status. <a href="https://getintoteaching.education.gov.uk/eligibility-for-teacher-training">Learn more</a>.</p>
```
Comment (tvararu) [inline app/views/candidate_interface/start_page/show.html.erb]: Isn't this content already written further down? 👀

Comment (alexdesi) [reply app/views/candidate_interface/start_page/show.html.erb]: good catch👌  I'm sure this was introduced with the rebase, I'll fix it!

**Pull request merged**